src/routes/(console)/project-[region]-[project]/sites/create-site/deploy/+page.svelte

```diff
@@ -0,0 +1,365 @@
+                type: 'error',
+                message: 'Domain is not valid'
+            });
+            return;
+        }
+
+        try {
+            // Create site with build configuration
+            let site = await sdk.forProject(page.params.region, page.params.project).sites.create(
+                id || ID.unique(),
+                name,
+                framework,
+                selectedFramework.buildRuntime,
+                undefined, // enabled
+                undefined, // logging
+                undefined, // timeout
+                installCommand || undefined,
+                buildCommand || undefined,
+                outputDirectory || undefined,
+                framework === Framework.Other ? Adapter.Static : undefined, // adapter
+                undefined, // installationId
+                undefined, // fallbackFile
+                undefined, // providerRepositoryId
+                undefined, // branch
+                false, // silentMode
+                rootDir || undefined
+            );
+
+            // Add domain
+            await sdk
+                .forProject(page.params.region, page.params.project)
+                .proxy.createSiteRule(
+                    `${domain}.${$regionalConsoleVariables._APP_DOMAIN_SITES}`,
+                    site.$id
+                );
+
+            // Add variables
+            const promises = variables.map((variable) =>
+                sdk
+                    .forProject(page.params.region, page.params.project)
+                    .sites.createVariable(site.$id, variable.key, variable.value, variable.secret)
+            );
+            await Promise.all(promises);
+
+            // Fetch tags from GitHub API
+            let latestTag = '';
+            try {
+                const tagsResponse = await fetch(
+                    `https://api.github.com/repos/${data.repository.owner}/${data.repository.name}/tags`
+                );
+                if (tagsResponse.ok) {
+                    const tags = await tagsResponse.json();
+                    if (tags.length > 0) {
+                        latestTag = tags[0].name;
+                    } else {
+                        addNotification({
+                            type: 'error',
+                            message:
+                                'No tags found in repository. Please create a tag before deploying.'
+                        });
+                        return;
+                    }
+                } else {
+                    addNotification({
+                        type: 'error',
+                        message: 'Failed to fetch tags from GitHub.'
+                    });
+                    return;
+                }
+            } catch (error) {
+                addNotification({
+                    type: 'error',
+                    message: 'Failed to fetch tags from GitHub: ' + error.message
+                });
+                return;
+            }
```

Comment:
lets cleanup by putting the GitHub based tasks in a separate helper function. @Meldiron do we have VCS API method to fetch a specific repo? I don't exactly remember. @atharvadeosthale If we have an SDK, no need for helper and use that.